pkg/util/bandwidth/interfaces.go

```diff
@@ -0,0 +1,23 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package bandwidth
+
+type BandwidthShaper interface {
+	InitializeInterface(iface string) error
+	LimitCIDRBandwidth(upload, download, cidr, iface string) error
```

Comment:
I guess `net_cls.classid` has the advantage that it can distinguish between two pods that use host networking, but the disadvantage that docker doesn't support setting it.